domain/src/main/java/org/sagebionetworks/research/domain/task/navigation/strategy/StepNavigationStrategy.kt

```diff
@@ -48,10 +49,11 @@ class StepNavigationStrategy {
         /**
          * NextStepStrategy step to navigate to based on the current task result.
          *
+         * @param task current task
          * @param taskResult current task result
          * @return identifier of next step, or null if this is the last step
          */
-        fun getNextStepIdentifier(taskResult: TaskResult?): String?
+        fun getNextStepIdentifier(task: Task, taskResult: TaskResult): String?
```

Comment:
A step should not need its Task. I see that this is still true in the iOS protocols as well